### DISCUSSION TITLE
PLEASE HELP!! Unable to compile with gcc 11.4.0

### DISCUSSION THREAD

**Discussion by princesswinnie1122** [ornladios/ADIOS2#4171] [Q&A]
Hello, I'm installing ADIOS2 for LAMMPS...but kept on failing the installation process.
Below are the details. Please let me know if I missing anything! Thanks a lot!

- The compiler I use (source ~/.bashrc)
    ```
    gcc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
    Copyright (C) 2021 Free Software Foundation, Inc.
    This is free software; see the source for copying conditions.  There is NO
    warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
    ```
    ```
    scteam06@head:~/lammps/build/ADIOS-2.8.0/build$ which mpicc
    /home/scteam06/openmpi/bin/mpicc
    ```
- Configuration:
    ```
    C=mpicc CXX=mpicxx FC=mpifort \
    cmake .. \
    -DMPI_Fortran_COMPILER=mpifort \
    -DCMAKE_Fortran_COMPILER=mpifort \
    -DCMAKE_C_COMPILER=mpicc \
    -DCMAKE_CXX_COMPILER=mpicxx \
    -DCMAKE_CXX_FLAGS="-O3 -fpic" \
    -DADIOS2_USE_MPI=ON \
    -DADIOS2_USE_Endian_Reverse=ON \
    -DADIOS2_USE_Fortran=ON \
    -DADIOS2_RUN_INSTALL_TEST=OFF \
    -DCMAKE_FIND_ROOT_PATH_MODE_LIBRARY=ONLY \
    -DCMAKE_FIND_ROOT_PATH_MODE_INCLUDE=ONLY \
    -DCMAKE_INSTALL_PREFIX=$BUILD/$ADIOS
    ```
    > Result log: 
    [cmake_configure.log](https://github.com/ornladios/ADIOS2/files/15398933/cmake_configure.log)

- **Installation (where I'm stuck)**
    ```
    make -j
    ```
    > Result log:
    [make_build.log](https://github.com/ornladios/ADIOS2/files/15398944/make_build.log)

- Dependencies I've built (all with openmpi also):
![image](https://github.com/ornladios/ADIOS2/assets/37247090/49f4f38b-3a82-4668-9ac8-d81afcd7edeb)

- How I set the paths:
     ```
    export HDF5_DIR=/home/scteam06/lammps/opt/hdf5-1.8.18
    export PNETCDF_DIR=/home/scteam06/lammps/opt/pnetcdf-1.12.3
    export NETCDF_DIR=/home/scteam06/lammps/opt/netcdf-c-4.8.1
    export NETCDF_FORTRAN_DIR=/home/scteam06/lammps/opt/netcdf-fortran-4.6.1
    
    export PATH=$HDF5_DIR/bin:$PNETCDF_DIR/bin:$NETCDF_DIR/bin:$PATH
    export LD_LIBRARY_PATH=$HDF5_DIR/lib:$PNETCDF_DIR/lib:$NETCDF_DIR/lib:$LD_LIBRARY_PATH
    export C_INCLUDE_PATH=$HDF5_DIR/include:$PNETCDF_DIR/include:$NETCDF_DIR/include:$C_INCLUDE_PATH
    export CPLUS_INCLUDE_PATH=$HDF5_DIR/include:$PNETCDF_DIR/include:$NETCDF_DIR/include:$CPLUS_INCLUDE_PATH
    export LIBRARY_PATH=$HDF5_DIR/lib:$PNETCDF_DIR/lib:$NETCDF_DIR/lib:$LIBRARY_PATH
    ```

**Comment by vicentebolea** [comment]
Adios 2.8 is now unmaintained and unsupported, is it possible to try adios 2.10 or 2.9?

> C=mpicc CXX=mpicxx FC=mpifort \
cmake .. \
-DMPI_Fortran_COMPILER=mpifort \
-DCMAKE_Fortran_COMPILER=mpifort \
-DCMAKE_C_COMPILER=mpicc \
-DCMAKE_CXX_COMPILER=mpicxx \
-DCMAKE_CXX_FLAGS="-O3 -fpic" \
-DADIOS2_USE_MPI=ON \
-DADIOS2_USE_Endian_Reverse=ON \
-DADIOS2_USE_Fortran=ON \
-DADIOS2_RUN_INSTALL_TEST=OFF \
-DCMAKE_FIND_ROOT_PATH_MODE_LIBRARY=ONLY \
-DCMAKE_FIND_ROOT_PATH_MODE_INCLUDE=ONLY \
-DCMAKE_INSTALL_PREFIX=$BUILD/$ADIOS

You can probably let go most of these options and remain only: 

```
-DADIOS2_USE_MPI=ON \
-DADIOS2_USE_Endian_Reverse=ON \
-DADIOS2_USE_Fortran=ON \
-DADIOS2_RUN_INSTALL_TEST=OFF \
-DCMAKE_INSTALL_PREFIX=$BUILD/$ADIOS

```

Also you should only need to set the _DIR vars for find_package. 

Also Gcc 11.4.0 is tricky, another version might help. If you just need adios2 installed you can use Spack to install it.
